unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxConfiguration.cs

```diff
@@ -0,0 +1,844 @@
+            {
+                if (percentage <= _list[i].percentage)
+                {
+                    max = _list[i];
+
+                    if ((i - 1) > 0)
+                    {
+                        min = _list[i - 1];
+                    }
+
+                    break;
+                }
+            }
+
+            float t = Mathf.InverseLerp(min.percentage, max.percentage, percentage);
+            return Mathf.Lerp(min.value, max.value, t);
+        }
+
+        Vector2 GetTransitionValue(List<TransitioningVector2> _list, float percentage, Vector2 defaultVal = default(Vector2))
+        {
+            Vector2 offset = defaultVal;
+
+            if (_list == null || _list.Count == 0)
+            {
+                return offset;
+            }
+
+            if (_list.Count == 1)
+            {
+                offset = _list[0].value;
+                return offset;
+            }
+
+
+            TransitioningVector2 min = _list[0], max = _list[0];
+
+            for (int i = 0; i < _list.Count; i++)
+            {
+                if (percentage <= _list[i].percentage)
+                {
+                    max = _list[i];
+
+                    if ((i - 1) > 0)
+                    {
+                        min = _list[i - 1];
+                    }
+
+                    break;
+                }
+            }
+            float t = Mathf.InverseLerp(min.percentage, max.percentage, percentage);
+            offset = Vector2.Lerp(min.value, max.value, t);
+
+            return offset;
+        }
+
+        Vector3 GetTransitionValue(List<TransitioningVector3> _list, float percentage)
+        {
+            Vector3 offset = new Vector3(0, 0, 0);
+
+            if (_list == null || _list.Count == 0)
+            {
+                return offset;
+            }
+
+            if (_list.Count == 1)
+            {
+                offset = _list[0].value;
+                return offset;
+            }
+
+
+            TransitioningVector3 min = _list[0], max = _list[0];
+
+            for (int i = 0; i < _list.Count; i++)
+            {
+                if (percentage <= _list[i].percentage)
+                {
+                    max = _list[i];
+
+                    if ((i - 1) > 0)
+                    {
+                        min = _list[i - 1];
+                    }
+
+                    break;
+                }
+            }
+
+            float t = Mathf.InverseLerp(min.percentage, max.percentage, percentage);
+            offset = Vector3.Lerp(min.value, max.value, t);
+
+            return offset;
+        }
+
+        #endregion
+
+        public void ResetMaterial(Material selectedMat, int slotCount)
+        {
+            for (int i = 0; i < slotCount; i++)
+            {
+                ResetSlot(selectedMat, i);
+            }
+        }
+
+        public void ResetSlot(Material selectedMat, int slotCount)
+        {
+            selectedMat.SetFloat("_layerType_" + slotCount, 0);
```

Comment:
This request is a big one, so let's organize for doing this on a new PR: All `SetFloat`, `SetVector`, `SetTexture` etc, should use the integer overload of the method, the string overloads are too slow when used aggresivelly like this. If the different slot versions for the uniform names are used we should precache them by using `Shader.PropertyToId("PropertyName")`.